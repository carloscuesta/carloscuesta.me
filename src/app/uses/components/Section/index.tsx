import SectionTitle from 'src/components/shared/SectionTitle'

type Props = {
  items: { category: string; link?: string; title: string }[]
  subTitle: string
  title: string
}

const Section = (props: Props) => (
  <section>
    <SectionTitle title={props.title} subTitle={props.subTitle} />

    <ul className="prose list-inside list-disc dark:prose-invert">
      {props.items.map((item) => (
        <li key={item.title}>
          {`${item.category} â€“ `}
          {item.link ? (
            <a href={item.link} rel="noopener noreferrer">
              {item.title}
            </a>
          ) : (
            item.title
          )}
        </li>
      ))}
    </ul>
  </section>
)

export default Section
