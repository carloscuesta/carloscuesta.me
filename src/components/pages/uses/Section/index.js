// @flow
import React, { type Element } from 'react'

import SectionTitle from 'src/components/shared/SectionTitle'
import styles from './styles.module.css'

type Props = {
  items: Array<{ category: string, link?: ?string, title: string }>,
  subTitle: string,
  title: string,
}

const Section = (props: Props): Element<'section'> => (
  <section>
    <SectionTitle title={props.title} subTitle={props.subTitle} />

    <ul className={styles.list}>
      {props.items.map((item) => (
        <li key={item.title}>
          <p className={styles.title}>
            {`${item.category} â€“ `}
            {item.link
              ? (
                <a href={item.link} rel='noopener noreferrer'>
                  {item.title}
                </a>
                )
              : item.title}
          </p>
        </li>
      ))}
    </ul>
  </section>
)

export default Section
