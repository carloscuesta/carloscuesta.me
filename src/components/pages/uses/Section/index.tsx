import SectionTitle from 'src/components/shared/SectionTitle'

type Props = {
  items: { category: string, link?: string, title: string }[],
  subTitle: string,
  title: string,
}

const Section = (props: Props) => (
  <section>
    <SectionTitle title={props.title} subTitle={props.subTitle} />

    <ul className='prose dark:prose-invert list-disc list-inside'>
      {props.items.map((item) => (
        <li key={item.title}>
          {`${item.category} â€“ `}
          {item.link
            ? (
              <a href={item.link} rel='noopener noreferrer'>
                {item.title}
              </a>
              )
            : item.title}
        </li>
      ))}
    </ul>
  </section>
)

export default Section
