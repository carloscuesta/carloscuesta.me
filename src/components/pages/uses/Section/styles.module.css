.list {
  padding: 0 1rem;
}

.title {
  margin: 0;
  margin-bottom: .5rem;
}
