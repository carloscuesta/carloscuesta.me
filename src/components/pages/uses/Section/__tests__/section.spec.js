import renderer from 'react-test-renderer'

import Section from '../index'
import * as stubs from './stubs'

describe('Section', () => {
  it('should match Section component', () => {
    const wrapper = renderer.create(<Section {...stubs.props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
