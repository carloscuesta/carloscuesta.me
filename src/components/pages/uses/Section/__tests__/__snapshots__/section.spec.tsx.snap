// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should match Section component 1`] = `
<section>
  <div
    className="container "
  >
    <div>
      <h2
        className="sectionTitle "
      >
        Gear
      </h2>
      <h3
        className="sectionSubtitle"
      >
        Home office equipment
      </h3>
    </div>
  </div>
  <ul
    className="list"
  >
    <li>
      <p
        className="title"
      >
        Desk – 
        <a
          href="https://www.fully.com/standing-desks/jarvis-adjustable-height-desk-bamboo.html"
          rel="noopener noreferrer"
        >
          Fully Jarvis Bamboo Standing Desk
        </a>
      </p>
    </li>
    <li>
      <p
        className="title"
      >
        Laptop – 
        <a
          href="https://support.apple.com/kb/sp704"
          rel="noopener noreferrer"
        >
          MacBook Pro 15 (Mid 2014), i7 @ 2,5GHz, 16 GB
        </a>
      </p>
    </li>
    <li>
      <p
        className="title"
      >
        Chair – 
        <a
          href="https://www.ikea.com/es/es/p/markus-silla-trabajo-glose-negro-40103100/"
          rel="noopener noreferrer"
        >
          Ikea Markus
        </a>
      </p>
    </li>
    <li>
      <p
        className="title"
      >
        Monitor – 
        Dell U2414H
      </p>
    </li>
  </ul>
</section>
`;
